### PR TITLE
feat: add file config in db

### PR DESCRIPTION
**Pull request by hetao92**
## What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

### CODE REVIEW

File: server/api/studio/internal/service/file.go [hunk 1]
```diff
@@ -33,43 +39,63 @@ var (
...
-	}
-
-	//	if target is directory, it is not empty
-	if err := os.Remove(target); err != nil {
-		logx.Infof("del file error %v", err)
-		return ecode.WithInternalServer(err)
+	for _, name := range request.Names {
+		target := filepath.Join(dir, name)
+		if _, err := os.Stat(target); err != nil {
+			logx.Infof("del file error %v", err)
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go:78]: `del`?

File: server/api/studio/internal/service/file.go [hunk 2]
```diff
@@ -33,43 +39,63 @@ var (
...
+			return ecode.WithInternalServer(err)
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go]: Maybe 404 suitable?

File: server/api/studio/internal/service/file.go [hunk 3]
```diff
@@ -88,36 +114,58 @@ func (f *fileService) FilesIndex() (data *types.FilesIndexData, err error) {
 		if fileInfo.IsDir() {
 			continue
 		}
+		var fileConfig db.File
+		result := db.CtxDB.Where("name = ?", fileInfo.Name()).First(&fileConfig)
+		if result.Error != nil {
+			logx.Errorf("get file config record in db error %v", result.Error)
+			fileConfig.Delimiter = ","
+			fileConfig.WithHeader = false
+		}
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go:123]: Why init with `","`, `false`?

File: server/api/studio/internal/service/file.go [hunk 4]
```diff
@@ -88,36 +114,58 @@ func (f *fileService) FilesIndex() (data *types.FilesIndexData, err error) {
...
 		}
-		reader := csv.NewReader(file)
+		reader := bufio.NewReader(file)
 		count := 0
-		content := make([][]string, 0)
-		for count < 3 {
-			line, err := reader.Read()
+		sample := ""
+		for count < 5 {
+			line, _, err := reader.ReadLine()
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go:133]: `for count:=0; count<5; count++` ?

File: server/api/studio/internal/service/file.go [hunk 5]
```diff
@@ -88,36 +114,58 @@ func (f *fileService) FilesIndex() (data *types.FilesIndexData, err error) {
...
 			count++
 			if err != nil {
 				break
 			}
-			content = append(content, line)
+			sample += string(line) + "\r\n"
 		}
+		file.Close()
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go:139]: 👍

File: server/api/studio/internal/service/file.go [hunk 6]
```diff
@@ -88,36 +114,58 @@ func (f *fileService) FilesIndex() (data *types.FilesIndexData, err error) {
...
+	result = db.CtxDB.Model(File).Updates(map[string]interface{}{"with_header": request.WithHeader, "delimiter": request.Delimiter})
+	if result.Error != nil {
+		return ecode.WithErrorMessage(ecode.ErrInternalDatabase, result.Error)
+	}
+	return nil
+}
 func (f *fileService) FileUpload() error {
 	dir := f.svcCtx.Config.File.UploadDir
+	auth := f.ctx.Value(auth.CtxKeyUserInfo{}).(*auth.AuthData)
+	host := auth.Address + ":" + strconv.Itoa(auth.Port)
```
Comment (veezhang) [inline server/api/studio/internal/service/file.go:183]: Seems unnecessary?

File: server/api/studio/internal/service/file.go (hunk 3 above)
Comment (hetao92) [reply server/api/studio/internal/service/file.go:123]: Considering that the user can directly upload the file to the corresponding directory of the server, the configuration record of the file may not exist in the database, so the default value is set and returned to the front end.

File: app/pages/Import/FileList/UploadConfigModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+import { ExclamationCircleFilled } from '@ant-design/icons';
+import styles from './index.module.less';
+interface IProps {
+  visible: boolean;
+  onConfirm: () => void;
+  onCancel: () => void;
+  uploadList: StudioFile[];
+}
+
+const DelimiterConfigModal = (props: { onConfirm: (string) => void }) => {
```
Comment (huaxiabuluo) [inline app/pages/Import/FileList/UploadConfigModal/index.tsx:22]: missing type, check all code in this commit

File: app/pages/Import/FileList/UploadConfigModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+    const newData = [...data];
+    newData.splice(index, 1);
+    const checkedList = newData.filter((file) => file.withHeader);
+    setState({
+      checkAll: checkedList.length === newData.length,
+      indeterminate: !!checkedList.length && checkedList.length < newData.length,
+      data: newData,
+      activeItem: isActive ? null : activeItem,
+      previewContent: isActive ? [] : previewContent,
+    });
```
Comment (huaxiabuluo) [inline app/pages/Import/FileList/UploadConfigModal/index.tsx]:
```ts
const newData = data.filter((_, i) => i !== index);
const checkedNum = newData.reduce((n, file) => n + (file.withHeader & 1), 0);

setState({
  checkAll: checkedNum === newData.length,
  indeterminate: !!checkedNum && checkedNum < newData.length,
  data: newData,
  activeItem: isActive ? null : activeItem,
  previewContent: isActive ? [] : previewContent,
});
```

File: app/components/CSVPreviewLink/index.tsx [hunk 9]
```diff
@@ -1,13 +1,15 @@
 import { Button, Popover, Table } from 'antd';
 import { } from 'antd/lib/button';
```
Comment (huaxiabuluo) [inline app/components/CSVPreviewLink/index.tsx]: delete

File: app/pages/Import/FileList/PreviewFileModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+import React, { useCallback, useEffect } from 'react';
+import { usePapaParse } from 'react-papaparse';
+import { StudioFile } from '@app/interfaces/import';
+import { observer } from 'mobx-react-lite';
+import { useBatchState } from '@app/utils';
+import { useStore } from '@app/stores';
+import styles from './index.module.less';
+interface IProps {
+  file: StudioFile;
+  children: any;
```
Comment (huaxiabuluo) [inline app/pages/Import/FileList/PreviewFileModal/index.tsx]:
```ts
type IProps = PropsWithChildren<{ file: StudioFile; }>

interface IProps {
  file: StudioFile;
  children: ReactNode;
}
```

File: app/pages/Import/FileList/index.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    if(global.gConfig?.maxBytes && size > global.gConfig.maxBytes) {
+      message.error(intl.get('import.fileSizeLimit', { size: getFileSize(global.gConfig.maxBytes) }));
+      return false;
+    }
+    fileList.forEach(file => {
+      file.path = `${file.name}`;
+      file.withHeader = false;
+      file.delimiter = ',';
+    });
+    setPreviewList(fileList);
```
Comment (huaxiabuluo) [inline app/pages/Import/FileList/index.tsx:84]:
```ts
setPreviewList(fileList.map(file => ({
  ...file,
  path: `${file.name}`,
  withHeader: false,
  delimiter: ',',
));
```

**Pull request merged**